### PR TITLE
interfaces/builtin/gpio_test.go: actually test the generated gpio apparmor

### PR DESCRIPTION
**Pull request by anonymouse64**
This was never actually being tested, and in addition the IgnoresMissingSymlink
test was evaluating symlinks on the host, and so the test would have failed if
run on a Raspberry Pi for example with gpio pins available.

Discovered while debugging https://github.com/snapcore/snapd/pull/10061

### CODE REVIEW

File: interfaces/builtin/gpio_test.go [hunk 1]
```diff
@@ -144,13 +151,31 @@ func (s *GpioInterfaceSuite) TestApparmorConnectedPlugIgnoresMissingSymlink(c *C
...
+	})
+
 	spec := &apparmor.Specification{}
 	err := spec.AddConnectedPlug(s.iface, s.gadgetPlug, s.gadgetGpioSlot)
 	c.Assert(err, IsNil)
 	c.Assert(spec.Snippets(), HasLen, 0)
 	c.Assert(log.String(), testutil.Contains, "cannot export not existing gpio /sys/class/gpio/gpio100")
 }
 
+func (s *GpioInterfaceSuite) TestApparmorConnectedSlot(c *C) {
```
Comment (jdstrand) [inline interfaces/builtin/gpio_test.go:166]: I think you meant `TestApparmorConnectedPlug`?

**Review by jdstrand — APPROVED** [review]
Other than the test rename, LGTM. Thanks!

File: interfaces/builtin/gpio_test.go (hunk 1 above)
Comment (anonymouse64) [reply interfaces/builtin/gpio_test.go:166]: ah yes indeed, good catch

**Review by mvo5 — APPROVED** [review]
Thank you!

**Pull request merged**